prov/efa/src/rdm/efa_rdm_ope.c

```diff
@@ -695,8 +699,10 @@ void efa_rdm_txe_handle_error(struct efa_rdm_ope *txe, int err, int prov_errno)
 	err_entry.data = txe->cq_entry.data;
 	err_entry.tag = txe->cq_entry.tag;
 	if (OFI_UNLIKELY(efa_rdm_write_error_msg(ep, txe->addr, prov_errno,
-	                                         &err_entry.err_data, &err_entry.err_data_size))) {
+	                                         err_msg, &err_entry.err_data_size))) {
 		err_entry.err_data_size = 0;
+	} else {
```

Comment:
Better to `err_entry.err_data = err_msg;` before you call `efa_rdm_write_error_msg`, so you have a valid `err_data` for ofi_cq_write_error anyway.

Reply:
Sorry, I think your current one may be better, just see  your err_entry.err_data is already NULL if the `efa_rdm_write_error_msg` returns non-zero.